starter-code/scripts/article.js

```diff
@@ -50,13 +50,25 @@ Article.fetchAll = function() {
     // When rawData is already in localStorage,
     // we can load it with the .loadAll function above,
     // and then render the index page (using the proper method on the articleView object).
-    Article.loadAll(); //TODO: What do we pass in to loadAll()?
-    //TODO: What method do we call to render the index page?
+    Article.loadAll(JSON.parse(localStorage.rawData)); //DONE: What do we pass in to loadAll()?
+    // est 5min act 5min
+    //DONE: What method do we call to render the index page?
+    // est: 5min act: 2min
+    articleView.initIndexPage();
   } else {
-    // TODO: When we don't already have the rawData,
+    // DONE: When we don't already have the rawData,
     // we need to retrieve the JSON file from the server with AJAX (which jQuery method is best for this?),
     // cache it in localStorage so we can skip the server call next time,
     // then load all the data into Article.all with the .loadAll function above,
     // and then render the index page.
+    //estimated time = 1hr Actual 1.5 hours
+    $.getJSON('../data/hackerIpsum.json').then(function (data) {
```

Comment:
When you have no local storage, your AJAX call does not work. I get your console.log(err) and a GET 404 Error.